### PR TITLE
binary-tree

### PR DESCRIPTION
**Pull request by Aarti002**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/602"></a>

### CODE REVIEW

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please add documentation in
all functions, classes and at the beginning.

For example, see [`conversions/int_to_string.c`](https://github.com/TheAlgorithms/C/blob/master/conversions/int_to_string.c).

**Comment by Aarti002** [comment]
Okay sir @Panquesito7 , i'll make sure to make that required changes , actually i'm a beginner to open source so i just want to explore as much as possible so i am glad you reviewed my commits thank u so much :)
Sorry for delay in reply!!!

**Comment by Aarti002** [comment]
sir @Panquesito7 i have added proper documentation have a look and let me know if something is still missing!

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
See [`conversions/int_to_string.c`](https://github.com/TheAlgorithms/C/blob/master/conversions/int_to_string.c) for how to document functions, variables, classes and the first comment block.

Also, please consider adding static testing functions.

**Comment by Aarti002** [comment]
Actually i'm little bit confused how to document.....?
I thought documentation is just explaining your functions its parameters, return type and stuff and also how its working.
And i tried my best to explain each function .I don't understand what else i need to add!!

**Comment by Aarti002** [comment]
And i already saw that attached file that you sent but i didn't get what's that @param  and @note stands for?
why can't we directly use return,param,note and  brief words without using @?

**Comment by Panquesito7** [comment]
It is because of the Doxygen documentation.
You need to do it using `@brief`, `@param`, `@returns`, `@details` and more.

See the [C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-Plus) repository [contributing guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md) for [file structure](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program). 🙂

**Comment by Aarti002** [comment]
ohhk 😀, then i will read that repository first.

**Comment by Aarti002** [comment]
i think now i have made appropriate changes please sir @Panquesito7 let me know if still something needs to add on !
And yes i want to tell i got to know so few new thinks while improving my files so thank you so much @Panquesito7 for your reviews!!!

File: data_structures/binary_trees/binarytree_approach2.c [hunk 1]
```diff
@@ -0,0 +1,270 @@
+/*
+@file 
+@brief have compelete implementation of a binary search tree
+@include all function defined 
+including insert,find,findMax,delet_tree,delete(to delete a node),preOrder and height.
+which is required for compelete implementation of a tree
```
Comment (Panquesito7) [inline data_structures/binary_trees/binarytree_approach2.c:6]: An `* ` is needed before any word is added in all comment blocks.
```suggestion
 * @file 
 * @brief have complete implementation of a binary search tree
 * @include all function defined 
 * including insert, find, findMax, delete_tree, delete(to delete a node), preOrder and height.
 * which is required for complete implementation of a tree
 *
 * @author [Aarti Singh](https://github.com/Aarti002)
```

File: data_structures/binary_trees/binarytree_approach2.c [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    if (root != NULL)
+    {
+    	printf("\t[ %d ]\t", root->data);
+        preOrder(root->left);
+        preOrder(root->right);
+    }
+}
+
+/** Main funcion 
+it has a menu (switch-cases) containg the list of function (defined above) to perform any particuler action*/
```
Comment (Panquesito7) [inline data_structures/binary_trees/binarytree_approach2.c:212]:
```suggestion
/**
 * @brief Main function
 * @details
 * Contains a menu (switch-cases) containing the list of function
 * (defined above) to perform any particular action
 *
 * @returns 0 on exit
 */
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Code looks much better, good work. 👍

I'd like you to make separate PR's for each file, so, each file is well-documented and refined. 🙂
We will also make the review process faster, which means we can get each PR merged quicker.

File: data_structures/binary_trees/binarytree_approach2.c (hunk 1 above)
Comment (Aarti002) [reply data_structures/binary_trees/binarytree_approach2.c:6]: i didnt get it ,how to make seperate PR for each file. :/

Comment (Panquesito7) [reply data_structures/binary_trees/binarytree_approach2.c:6]: Make a PR per the files you've edited in this PR.

**Comment by Aarti002** [comment]
Now i have made separate PR for my two files, i hope you would merge them now if there is no more conflicts :)
Thank you!!!

**Comment by Aarti002** [comment]
I don't know both are separated or not, yesterday i made PR for one file and today for another .
Let me know they are separate or not!

**Comment by Panquesito7** [comment]
I still don't see any other PRs here. There are 3 files being modified in this PR. By the way, would you like to continue working on this PR? 🙂

**Pull request closed without merging**